### PR TITLE
Resolve gnarly looping bug for nodes with AWAIT ATTRIBUTES merge behavior

### PR DESCRIPTION
**Pull request by dvargas92495**
We have an issue with AWAIT_ANY nodes where loops in the middle of a graph never execute. Turns out this problem _also_ existed for AWAIT_ATTRIBUTES node.

It should've never existed for the latter - those nodes only execute based on their attribute definitions. This PR moves the `queue_node_execution` into the `BaseNode.Trigger` class so that we can discriminate that step based on merge behavior. This solves this gnarly edge case for `AWAIT_ATTRIBUTES` nodes while deferring our need to fix it for AWAIT_ANY node in the future

### CODE REVIEW

File: src/vellum/workflows/nodes/bases/base.py [hunk 1]
```diff
@@ -325,6 +326,36 @@ def should_initiate(
...
+            We may combine this into the should_initiate method, but we'll keep it separate for now to avoid
+            breaking changes until the 0.15.0 release.
+            """
+
+            execution_id = uuid4()
+            if not invoked_by:
+                return execution_id
+
+            if cls.merge_behavior not in {MergeBehavior.AWAIT_ANY, MergeBehavior.AWAIT_ALL}:
+                return execution_id
```
Comment (dvargasfuertes) [inline src/vellum/workflows/nodes/bases/base.py:345]: This is the only new piece of logic here. every thing else is copy pasted. The doc string for the method is also new

File: src/vellum/workflows/runner/runner.py [hunk 2]
```diff
@@ -431,7 +431,7 @@ def _run_node_if_ready(
                     return
 
             all_deps = self._dependencies[node_class]
-            node_span_id = state.meta.node_execution_cache.queue_node_execution(node_class, all_deps, invoked_by)
+            node_span_id = node_class.Trigger._queue_node_execution(state, all_deps, invoked_by)
             if not node_class.Trigger.should_initiate(state, all_deps, node_span_id):
```
Comment (dvargasfuertes) [inline src/vellum/workflows/runner/runner.py:435]: I imagine combining these two in a future release

File: tests/workflows/basic_await_any/workflow.py [hunk 3]
```diff
@@ -25,6 +26,9 @@ def run(self) -> Outputs:
 class AwaitAnyNode(BaseNode[BaseState]):
     top = TopNode.Outputs.total
 
+    class Trigger(BaseNode.Trigger):
+        merge_behavior = MergeBehavior.AWAIT_ANY
```
Comment (dvargasfuertes) [inline tests/workflows/basic_await_any/workflow.py:30]: This test was always busted. It claimed to be an AwaitAny node, but was still using the default AWAIT_ATTRIBUTES strategy

File: tests/workflows/basic_await_attributes/tests/test_workflow.py [hunk 4]
```diff
@@ -14,4 +14,4 @@ def test_workflow__happy_path():
 
     # THEN the Workflow completes successfully
     assert final_event.name == "workflow.execution.fulfilled"
-    assert final_event.outputs.final_value == 2
+    assert final_event.outputs.final_value == 3
```
Comment (dvargasfuertes) [inline tests/workflows/basic_await_attributes/tests/test_workflow.py:17]: This test was also busted. An AWAIT_ATTRIBUTES node should be invoked three times if there are three dependencies and the attributes are set for all three

File: tests/workflows/basic_emitter_workflow/tests/test_workflow.py [hunk 5]
```diff
@@ -25,8 +25,9 @@ def test_run_workflow__happy_path(mock_uuid4_generator, mock_datetime_now):
     state_id_generator = mock_uuid4_generator("vellum.workflows.state.base.uuid4")
     state_id = state_id_generator()
     workflow_span_id = state_id_generator()
-    start_node_span_id = state_id_generator()
-    next_node_span_id = state_id_generator()
+    base_node_id_generator = mock_uuid4_generator("vellum.workflows.nodes.bases.base.uuid4")
```
Comment (dvargasfuertes) [inline tests/workflows/basic_emitter_workflow/tests/test_workflow.py:28]: bc we moved the method, we needed to move the fixture

File: tests/workflows/basic_emitter_workflow/tests/test_workflow.py [hunk 6]
```diff
@@ -120,12 +121,8 @@ def test_run_workflow__happy_path(mock_uuid4_generator, mock_datetime_now):
...
                     serialized_next_node_id: [str(next_node_span_id)],
                 },
-                "node_executions_queued": {
-                    serialized_next_node_id: [],
-                },
-                "dependencies_invoked": {
-                    str(next_node_span_id): [serialized_start_node_id],
-                },
+                "node_executions_queued": {},
+                "dependencies_invoked": {},
```
Comment (dvargasfuertes) [inline tests/workflows/basic_emitter_workflow/tests/test_workflow.py:125]: we still consider the `node_executrion_cache` to be an unstable API, and we should feel free to mess with it

File: tests/workflows/circular_loop/tests/test_workflow.py [hunk 7]
```diff
@@ -1,25 +1,56 @@
...
+
+    assert terminal_event.outputs.counter == 2
+    assert terminal_event.outputs.completed is True
+
+
 @pytest.mark.skip(
     reason="https://linear.app/vellum/issue/APO-390/workflow-erronously-terminates-in-looping-edge-case-with-prefixed-node"  # noqa: E501
 )
-def test_workflow__happy_path():
+def test_workflow__top_node_awaits_any():
```
Comment (dvargasfuertes) [inline tests/workflows/circular_loop/tests/test_workflow.py:31]: we still want to track and fix this issue for AWAIT ANY nodes one day

File: tests/workflows/multi_branch_merge_loop/workflow.py [hunk 8]
```diff
@@ -49,6 +50,9 @@ class Ports(BaseNode.Ports):
     class Outputs(BaseNode.Outputs):
         value = TopNode.Outputs.value.coalesce(BottomNode.Outputs.value)
 
+    class Trigger(BaseNode.Trigger):
+        merge_behavior = MergeBehavior.AWAIT_ANY
```
Comment (dvargasfuertes) [inline tests/workflows/multi_branch_merge_loop/workflow.py:54]: Another test that was assumed to be an AWAIT ANY node but wasn't under the hood

**Comment by dvargas92495** [comment]
@vincent0426 with this, you should be able to iterate on ToolCalling Node
